### PR TITLE
https://tokbox.atlassian.net/browse/ECO-20

### PR DESCRIPTION
**Pull request by hbaqai**
All sample code for the Basics tutorial should work in this branch. I have modified the README to have point to the branch with completed code instead of having each step for now. Since we're working towards that goal I felt it made sense to just do it now. We still need to modify the tutorial on the developer center to match with this approach.

### CODE REVIEW

File: app/src/main/java/com/tokbox/android/demo/learningopentok/ChatActivity.java [hunk 1]
```diff
@@ -70,13 +72,13 @@ public boolean onOptionsItemSelected(MenuItem item) {
     }
 
     private void initializeSession() {
-        mSession = new Session(this, mApiKey, mSessionId);
+        mSession = (new Session.Builder(this, mApiKey, mSessionId).build());
```
Comment (jeffswartz) [inline app/src/main/java/com/tokbox/android/demo/learningopentok/ChatActivity.java]: I think we should remove a pair of parens:

``` mSession = new Session.Builder(this, mApiKey, mSessionId).build();```

File: app/src/main/java/com/tokbox/android/demo/learningopentok/ChatActivity.java [hunk 2]
```diff
@@ -70,13 +72,13 @@ public boolean onOptionsItemSelected(MenuItem item) {
...
         mSession.setSessionListener(this);
         mSession.connect(mToken);
     }
 
     private void initializePublisher() {
-        mPublisher = new Publisher(this);
+        mPublisher = (new Publisher.Builder(this)).build();
```
Comment (jeffswartz) [inline app/src/main/java/com/tokbox/android/demo/learningopentok/ChatActivity.java]: I think we should remove a pair of parens:

``` mPublisher = new Publisher.Builder(this).build();```

(Note that you group the additional parens differently here than you do in the `Session.Builder` line.)

File: app/src/main/java/com/tokbox/android/demo/learningopentok/ChatActivity.java [hunk 3]
```diff
@@ -126,7 +128,7 @@ public void onStreamReceived(Session session, Stream stream) {
         Log.i(LOG_TAG, "Stream Received");
 
         if (mSubscriber == null) {
-            mSubscriber = new Subscriber(this, stream);
+            mSubscriber = (new Subscriber.Builder(this, stream)).build();
```
Comment (jeffswartz) [inline app/src/main/java/com/tokbox/android/demo/learningopentok/ChatActivity.java]: Again: remove the parens:

```mSubscriber = new Subscriber.Builder(this, stream).build();```

Comment (jeffswartz) [reply app/src/main/java/com/tokbox/android/demo/learningopentok/ChatActivity.java]: I didn't realize we have a Subscriber.Builder. Good catch.

**Review by jeffswartz — COMMENTED** [review]
All of my comments are optional.

File: README.md [hunk 4]
```diff
@@ -23,13 +23,7 @@ https://tokbox.com/developer/tutorials/android/.
...
 The code for this sample is found the following git branches:
 
-* *basics.step-1* -- This branch shows you how to set up your project to use the OpenTok Android SDK.
-
-* *basics.step-4* -- This branch shows you how to connect to the OpenTok session.
-
-* *basics.step-5* -- This branch shows you how publish a stream to the OpenTok session.
-
-* *basics.step-6* -- This branch shows you how to subscribe to a stream on the OpenTok session.
+* *basics.completed* -- This branch contains a completed version of the [Basic Video Chat Tutorial](https://tokbox.com/developer/tutorials/android/basic-video-chat/).
```
Comment (jeffswartz) [inline README.md]: Let's just call this branch basics (not basics.completed).

**Comment by hbaqai** [comment]
Done

**Pull request merged**